src/swap/SwapScreen.tsx

```diff
@@ -46,15 +46,23 @@ export function SwapScreen() {
   const { t } = useTranslation()
   const dispatch = useDispatch()
 
-  const coreTokens = useSelector(coreTokensSelector)
+  const supportedTokens = useSelector(coreTokensSelector)
+  const supportedTokensAddresses = supportedTokens.map((token) => token.address)
```

Comment:
this is only used in one place, so maybe we don't need to declare it here?

variables are generally recreated when the component rerenders so depending on how big the list of supported tokens is, it might be something we want to wrap in `useMemo` to avoid doing the loop all the time. it doesn't matter much right now with 4 tokens of course, it just got me thinking that maybe we don't need this variable, and instead can use `supportedTokens` in `defaultFromToken` and wrap `defaultFromToken` in a `useMemo`. what do you think?

Reply:
Yep, that makes a lot of sense, I'll do that :)